### PR TITLE
Csv export 333

### PR DESCRIPTION
**Pull request by wgilling**
This feels pretty big and the likelihood that it needs more work on some fields is pretty high in my estimation, but it's time to get the merge process started with a review.

First note is that all multi-field delimiters have been changed from the "||" to " || " so that items with missing parts such as no subtitle on a complex title paragraph may be imported without confusion. The migrate_plus.migration.csv.yml has this delimiter coded into those fields, so I only mention this because the previous migrations from legacy CSV had the delimiters with no spaces around them.

The view `rest_csv_export` provides three different displays -- one for single item, collection, and search. These all have endpoints that make it fairly easy to manually download any given CSV export. For example, and item that is at `items/12345` has the route `items/12345/csv` and a collection that has `collections/123` has the csv route at `collections/123/csv` and any search pattern should be supported by adding the /csv to the end such as `search/csv?search_api_fulltext=data`. There should be a link in the Admin Toolbox to Download CSV from the items and collections page and another link on search pages near the "Export MODS XML".

The code uses Migrate Source UI in order to allow for a web interface for importing any CSV that was exported and modified. This is at http://localhost:8000/migrate_source_ui and is "Import/Update nodes from CSV (supports csv)".

To test, it seems that you know what to do... but it can be a fairly long process to set up alternate field vocabulary for testing purposes. When exporting, I found that it saves a lot of time to use a collection that has less than 5 items... not that the export takes a long time to run, but it is much easier to inspect and edit a smaller CSV file.

The import csv configurations will need to be imported. These are at `web/modules/custom/asu_migrate/config/install/migrate_plus.migration.csv.yml`.

For a command-line import test, the CSV file that is stored at web/modules/custom/asu_migrate/data/csv.csv can be overwritten in order to just use the `drush migrate:import csv --update` command.

Any questions, please ask.

### CODE REVIEW

File: web/modules/custom/asu_migrate/src/Plugin/Field/FieldFormatter/VisibilityCSVFormatter.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $elements = [];
+    $asu_utils = \Drupal::service('asu_utils');
+    // This takes each node and converts the moderation_state of it into the
+    // Visibility value.
+    //  - Private: draft
+    //  - Public: published
+    foreach ($items as $delta => $item) {
+      $item_entity_id = $item->value;
+      $item_entity = @\Drupal::entityTypeManager()->getStorage('node')->load($item_entity_id);
```
Comment (elizoller) [inline web/modules/custom/asu_migrate/src/Plugin/Field/FieldFormatter/VisibilityCSVFormatter.php]: dependency injection?

File: web/modules/custom/asu_admin_toolbox/src/Plugin/Block/AdminToolboxBlock.php [hunk 2]
```diff
@@ -141,7 +142,11 @@ public function build() {
     $node = $this->routeMatch->getParameter('node');
     $is_collection = ($node->bundle() == 'collection');
     $is_complex_object = FALSE;
-    if ($node->bundle() == 'asu_repository_item') {
+    $is_asu_repository_item = ($node->bundle() == 'asu_repository_item');
+    $user = User::load(\Drupal::currentUser()->id());
```
Comment (elizoller) [inline web/modules/custom/asu_admin_toolbox/src/Plugin/Block/AdminToolboxBlock.php]: dependency injection?

File: web/modules/custom/asu_migrate/config/install/migrate_plus.migration.c270_node.yml [hunk 3]
```diff
@@ -0,0 +1,522 @@
+# Uninstall this config when the feature is uninstalled
```
Comment (elizoller) [inline web/modules/custom/asu_migrate/config/install/migrate_plus.migration.c270_node.yml]: can we take this file out of this PR?

**Comment by wgilling** [comment]
Addressed the dependency injection issues and removed the c270 file. There were four process plugins that actually needed entityTypeManager to use dependency injection. There was also one bit in the AdminToolboxBlock code that needed dependency injection to replace the User:: call.

Please let me know if there are any issues when reviewing.

**Comment by wgilling** [comment]
Re-ready to re-review.

File: web/modules/custom/asu_migrate/config/install/migrate_plus.migration.csv.yml [hunk 4]
```diff
@@ -0,0 +1,513 @@
...
+
+process:
+  nid: ID
+    
+  field_extent:
+    plugin: skip_on_empty
+    source: Extent
+    method: process
+
+# wgg - Date Created should not be the EDTF field.
```
Comment (elizoller) [inline web/modules/custom/asu_migrate/config/install/migrate_plus.migration.csv.yml]: why is this commented out?

File: web/modules/custom/asu_migrate/src/Plugin/Field/FieldFormatter/VisibilityCSVFormatter.php [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $elements = [];
+    $asu_utils = \Drupal::service('asu_utils');
```
Comment (elizoller) [inline web/modules/custom/asu_migrate/src/Plugin/Field/FieldFormatter/VisibilityCSVFormatter.php]: :D dependency injection?

**Review by elizoller — CHANGES_REQUESTED** [review]
I tested the export on several levels (search, collection, item) and after the minor tweaks i pushed up, it seems ok for the collections i have in my local. 

i haven't tested the import back in feature, but did see one thing that'll need addressing which i flagged (the missing date created field)

**Comment by wgilling** [comment]
The latest commit was tested to export and import Created Date and correctly update the node value for that EDTF field.

**Review by elizoller — APPROVED** [review]

**Pull request merged**